llvmlite/binding/ffi.py

```diff
@@ -77,28 +79,96 @@ def __exit__(self, *exc_details):
         self._lock.release()
 
 
+class _suppress_cleanup_errors:
+    def __init__(self, context):
+        self._context = context
+
+    def __enter__(self):
+        return self._context.__enter__()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        try:
+            return self._context.__exit__(exc_type, exc_value, traceback)
+        except PermissionError:
+            pass  # Resource dylibs can't be deleted on Windows.
+
+
 class _lib_wrapper(object):
     """Wrap libllvmlite with a lock such that only one thread may access it at
     a time.
 
     This class duck-types a CDLL.
     """
-    __slots__ = ['_lib', '_fntab', '_lock']
 
-    def __init__(self, lib):
-        self._lib = lib
-        self._fntab = {}
+    def __init__(self):
         self._lock = _LLVMLock()
+        self._lib_handle = None
+
+    def _load_lib(self):
+        _lib_load_errors = []
+        # llvmlite native code may exist in one of two places, with the first
+        # taking priority:
+        # 1) In a shared library available as a resource of the llvmlite
+        #    package. This may involve unpacking the shared library from an
+        #    archive.
+        # 2) Linked directly into the main binary. Symbols may be resolved
+        #    from the main binary by passing None as the argument to
+        #    ctypes.CDLL.
+        for lib_context in (
+                _suppress_cleanup_errors(importlib.resources.path(
+                    __name__.rpartition(".")[0], get_library_name())),
+                contextlib.nullcontext(None)):
+            try:
+                with lib_context as lib_path:
+                    if os.name == 'nt' and not lib_path:
+                        # The Windows implementation of ctypes.CDLL does not
```

Comment:
Question here: does this mean that Windows doesn't support the kind of static linking?